manifests/params.pp

```diff
@@ -32,29 +32,55 @@
       }
     }
     'RedHat': {
-      $dnsdir             = '/etc'
-      $vardir             = '/var/named'
-      $optionspath        = '/etc/named/options.conf'
-      $zonefilepath       = "${vardir}/dynamic"
-      $localzonepath      = "${dnsdir}/named.rfc1912.zones"
-      $defaultzonepath    = 'unmanaged'
-      $publicviewpath     = "${dnsdir}/named/zones.conf"
-      $viewconfigpath     = "${dnsdir}/named/views"
-      $dns_server_package = 'bind'
-      $namedservicename   = 'named'
-      $user               = 'named'
-      $group              = 'named'
-      $rndcconfgen        = '/usr/sbin/rndc-confgen'
-      $named_checkconf    = '/usr/sbin/named-checkconf'
-      $sysconfig_file     = '/etc/sysconfig/named'
+      $user = 'named'
+      $group = 'named'
       $sysconfig_template = "dns/sysconfig.${facts['os']['family']}.erb"
       $sysconfig_startup_options = undef
       $sysconfig_disable_zone_checking = undef
 
       # This option is not relevant for RedHat
       $sysconfig_resolvconf_integration = undef
 
-      $dnssec_enable = if versioncmp($facts['os']['release']['major'], '9') >= 0 { undef } else { 'yes' }
+      if $dns::globals::scl {
+        $sclroot = '/opt/isc/isc-bind/root'
+        $sclconfroot = '/etc/opt/isc/scls/isc-bind'
+        $sclenvname = 'isc-bind'
```

Comment:
Perhaps I'm making it too dynamic, but consider:
```suggestion
        $sclvendor = 'isc'
        $sclenvname = "${sclvendor}-bind"
        $sclroot = "/opt/${sclvendor}/${sclenvname}/root"
        $sclconfroot = "/etc/opt/${sclvendor}/scls/${sclenvname}"
```

You could then even consider changing `$dns::globals::scl` to `Optional[Pattern[/^.+-.+$/]] $scl = undef`. You then only need to extract `$sclvendor` from `$scl` by splitting the string and it becomes possible to use different SCL too. Not sure if they exist so feel free to ignore that.

Reply:
split variant has been implemented